src/components/BattleRoom/index.js

```diff
@@ -58,6 +80,31 @@ export default function BattleRoom() {
     getSong();
   }, [roomId]);
 
+  useEffect(() => {
+    if (socket) {
+      socket.on(UPDATE_USER, (currentUserList) => {
+        setCurrentUserList(currentUserList);
+      });
+    }
+  }, [socket]);
+
+  useEffect(() => {
+    if (socket) {
+      socket.on(USER_JOINED, (user) => {
+        setCurrentUserList((prevUserList) => [...prevUserList, user]);
+      });
+    }
+  }, [socket]);
+
+  useEffect(() => {
+    if (socket) {
+      socket.on(USER_LEAVE, (user) => {
+        setCurrentUserList((prevUserList) =>
+          prevUserList.filter((users) => users.uid !== user.uid),
+        );
+      });
+    }
+  }, [socket]);
```

Comment:
`socket`에서 정보를 받아오는 같은 결의 `useEffect`가  실행문이 여러번 쓰여 있네요. 
가독성 면에서 하나로 중첩해주는 것이 어떨까 싶습니다. 같은 `dependency`를 가졌기에 합쳐줄 수 있을 것 같습니다.
```suggestion
useEffect(() => {
  if (socket) {
    socket.on(UPDATE_USER, (currentUserList) => {
      setCurrentUserList(currentUserList);
    });
    socket.on(USER_JOINED, (user) => {
      setCurrentUserList((prevUserList) => [...prevUserList, user]);
    });
    socket.on(USER_LEAVE, (user) => {
      setCurrentUserList((prevUserList) =>
        prevUserList.filter((users) => users.uid !== user.uid),
      );
    });
  }
}, [socket]);
```